8367613: Test compiler/runtime/TestDontCompileHugeMethods.java failed

caoman opened the pull request:
Hi,

Could anyone approve this change that exclude this test when running with `-Xcomp`? This avoids the test failure reported in [JDK-8367613](https://bugs.openjdk.org/browse/JDK-8367613).

For reasons I don't yet understand, the `HugeSwitch::shortMethod` method is not compiled under `-Xcomp  -XX:TieredStopAtLevel=1`. The method gets compiled with either `-Xcomp` or `-XX:TieredStopAtLevel=1`, but not both. I appreciate if anyone could provide insights on possible reasons.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8367613](https://bugs.openjdk.org/browse/JDK-8367613): Test compiler/runtime/TestDontCompileHugeMethods.java failed (**Bug** - P4)


### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Damon Fenacci](https://openjdk.org/census#dfenacci) (@dafedafe - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/27306/head:pull/27306` \
`$ git checkout pull/27306`

Update a local copy of the PR: \
`$ git checkout pull/27306` \
`$ git pull https://git.openjdk.org/jdk.git pull/27306/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 27306`

View PR using the GUI difftool: \
`$ git pr show -t 27306`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/27306.diff">https://git.openjdk.org/jdk/pull/27306.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/27306#issuecomment-3295996483)
</details>

dafedafe reviewed (commented):
Marginal thing: since the issue happens with `-Xcomp` and `-XX:TieredStopAtLevel=1` it might be good to add the latter to `@requires` to restrict it as much as possible.
Also you might want to add this bug to the `@bug` tag.

caoman commented:
Thank you for the review and suggestions.

@chhagedorn Thank you for the explanation. I switched to `-XX:CompileCommand=dontinline` for `shortMethod()`. It works for `-Xcomp -XX:TieredStopAtLevel=1`.

The benefit of `dontinline` approach is that it allows the test run under `-Xcomp`, esp. `-XX:-TieredCompilation` with `-Xcomp`. It is also future-proof, in case C2 manages to inline `shortMethod()` into `main()` under `-Xcomp` in the future.

Also added bug number. And excluding `-XX:TieredStopAtLevel=1` is no longer needed.

dafedafe reviewed (approved):
Thanks @caoman. LGTM

caoman commented:
/integrate

caoman commented:
@chhagedorn Could you also approve the latest commit?
